src/mongo/db/storage/tokuft/tokuft_dictionary.cpp

```diff
@@ -80,13 +81,13 @@ namespace mongo {
         }
 
         int _getWriteFlags(OperationContext *opCtx, bool skipPessimisticLocking=false) {
-            return (skipPessimisticLocking || _isReplicaSetSecondary(opCtx))
+            return (skipPessimisticLocking || _isReplicaSetSecondary(opCtx) || opCtx->HasBounds())
                     ? DB_PRELOCKED_WRITE
                     : 0;
         }
 
         int _getReadFlags(OperationContext *opCtx, bool skipPessimisticLocking) {
-            return (skipPessimisticLocking || _getDBTxn(opCtx).is_read_only() || _isReplicaSetSecondary(opCtx))
+            return (skipPessimisticLocking || _getDBTxn(opCtx).is_read_only() || _isReplicaSetSecondary(opCtx) || opCtx->HasBounds())
```

Comment:
Here `is_read_only()` turns `DB_PRELOCKED_WRITE` on. But in `OperationShouldPrelockCursor` function the same `is_read_only()` value means 'do not prelock cursor'. Is this ok?